cdap-docs/developer-manual/source/pipelines/developing-plugins/plugin-basics.rst

```diff
@@ -126,3 +126,75 @@ In this example, we have a plugin of type ``batchsource``, named ``MyBatchSource
 This plugin takes two configuration properties. The first is named ``input-path`` and is required.
 The second is named ``cleanOutput`` and is optional. Note that optional configuration fields should
 have their default values set in the no-argument constructor.
+
+Plugin Data Types
+=================
+
+Plugins support several physical data types: null, boolean, int, long, float, double, string, array, map, enum, union,
+and record. They also support several logical data types: date, time-millis, time-micros, timestamp-millis,
+and timestamp-micros. Logical types represent some abstract concept but are internally represented with a corresponding
+physical type. The date logical type is an integer that holds the number of days since the Unix epoch. The time type
+can be an int that holds milliseconds after midnight. Time type can also be a long that holds microseconds after
```

Comment:
The time-millis logical type is an integer that holds the number of milliseconds since midnight. The time-micros logical type is a long that holds the number of microseconds since midnight. The timestamp-millis logical type is a long that holds the number of milliseconds since the Unix epoch. The timestamp-micros logical type is a long that holds the number of microseconds since the Unix epoch.

And remove the 'Time type can also...' sentence.